File: api/v1/servicios/articulos/src/infraestructura/dto/ArticuloDTO.js
class ArticuloDTO {
  constructor(modeloMongoose) {
    this.id = modeloMongoose._id;
    this.titulo = modeloMongoose.titulo;
    this.juego = modeloMongoose.juego;
    this.categoria = modeloMongoose.categoria;
    this.descripcion = modeloMongoose.descripcion;
    this.contenido = modeloMongoose.contenido;
    this.imagen = modeloMongoose.imagen || '';
    this.fechaCreacion = modeloMongoose.fechaCreacion;
    this.fechaActualizacion = modeloMongoose.fechaActualizacion;
  }
}

exports.ArticuloDTO = ArticuloDTO;

File: Front/MetroidWikiClient/src/main/java/com/metroidwiki/model/ArticuloDTO.java
package com.metroidwiki.model;

public class ArticuloDTO {
    private String id;
    private String titulo;
    private String categoria;
    private String descripcion;
    private String contenido;
    private String estado;
    private String imagen;
    private int vistas;
    private String fechaCreacion;
    private String autor;

    public ArticuloDTO() {}

    public ArticuloDTO(String titulo, String categoria, String descripcion, String contenido, String estado, String autor) {
        this.titulo = titulo;
        this.categoria = categoria;
        this.descripcion = descripcion;
        this.contenido = contenido;
        this.estado = estado;
        this.autor = autor;
    }

    // Getters
    public String getId() { return id; }
    public String getTitulo() { return titulo; }
    public String getCategoria() { return categoria; }
    public String getDescripcion() { return descripcion; }
    public String getContenido() { return contenido; }
    public String getEstado() { return estado; }
    public String getImagen() { return imagen; }
    public int getVistas() { return vistas; }
    public String getFechaCreacion() { return fechaCreacion; }
    public String getAutor() { return autor; }

    // Setters necesarios
    public void setImagen(String imagen) { this.imagen = imagen; }
    public void setEstado(String estado) { this.estado = estado; }
    public void setVistas(int vistas) { this.vistas = vistas; }
    public void setFechaCreacion(String fechaCreacion) { this.fechaCreacion = fechaCreacion; }
    public void setAutor(String autor) { this.autor = autor; }
}